File: Java/data-ordering-system/src/functions/python/buscas.py
import time

def linearSearch(arr, search):
        
        startTime = time.time()
        
        n = len(arr)
        
        found = ""
        
        for i in range(n):
            
            if(str(arr[i]) == search):
                found = arr[i]
                break
            
        
        endTime = time.time()
        
        global totalTime
        
        totalTime = endTime - startTime
        
        return found
    
def binarySearch(arr, search):
        
        startTime = time.time()
        
        inicio = 0
        fim = len(arr)-1
        
        meio = 0
        
        found = ""
        
        while (inicio <= fim):
        
            meio = (inicio + fim) / 2

            if(arr[meio] == search):
                found = arr[meio]
                break
            
            if(arr[meio] > search):
                fim = meio - 1
            else:
                inicio = meio + 1
                
        endTime = time.time()
        
        global totalTime
        
        totalTime = endTime - startTime
        
        return found

def totalTimeFunc():
    return totalTime

File: Java/data-ordering-system/src/functions/java/Buscas.java
package src.functions.java;

import java.util.ArrayList;

import javax.swing.JFrame;
import javax.swing.JOptionPane;

public class Buscas {
    
    private String found;

    private int index;

    public void buscaLinear(ArrayList<?> arr, String value, JFrame frame){

        for (int i = 0; i < arr.size(); i++) {
            
            if(arr.get(i).toString().equals(value)){
                found = arr.get(i).toString();
                index = i;
                break;
            }
            
        }

        if(found != null){
            JOptionPane.showMessageDialog(frame, "Encontrado: " + found + " na posição: " + index);
        }else{
            JOptionPane.showMessageDialog(frame, "Valor: " + value + " não encontrado");
        }
    }   

    public void buscaBinaria(ArrayList<?> arr, String value, JFrame frame){

        int inicio = 0, fim = arr.size()-1;
        
        int meio;
        
        while (inicio <= fim) {
        
            meio = (inicio + fim) / 2;
        
            if (arr.get(meio).toString().equals(value)){
                found = arr.get(meio).toString();
                index = meio;
            }
        
            if (arr.get(meio).toString().compareToIgnoreCase(value) > 0){
            
                fim = meio - 1;

            } else {
        
                inicio = meio + 1;
            }

        }
        
        if(found != null){
            JOptionPane.showMessageDialog(frame, "Encontrado: " + found + " na posição: " + index);
        }else{
            JOptionPane.showMessageDialog(frame, "Valor: "+ value + " não encontrado!\nVerifique se a lista está ordenada, pela colula selecionada!");
        }
    }

}
